opentracing/span.h

```diff
@@ -1,200 +1,119 @@
-     * system), and with it some powerful costs: use this feature with care.
-     *
-     * IMPORTANT NOTE #1: `setBaggageItem()` will only propagate trace
-     * baggage items to *future* children of the `Span`.
-     *
-     * IMPORTANT NOTE #2: Use this thoughtfully and with care. Every key and
-     * value is copied into every local *and remote* child of this `Span`, and
-     * that can add up to a lot of network and CPU overhead.
-     *
-     * IMPORTANT NOTE #3: Baggage item keys have a restricted format:
-     * implementations may wish to use them as HTTP header keys (or key
-     * suffixes), and of course HTTP headers are case insensitive.
-     *
-     * As such, `restrictedKey` MUST match the regular expression
-     * `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. That is, it must
-     * start with a letter or number, and the remaining characters must be
-     * letters, numbers, or hyphens. See `canonicalizeBaggageKey()`. If
-     * `restrictedKey` does not meet these criteria, `setBaggageItem()`
-     * results in undefined behavior.
-     *
-     * Returns a pointer to this `Span` for chaining, etc.
-     */
-    virtual Span * setBaggageItem(const std::string & restrictedKey, const std::string & value);
-
-    /*
-     * If `targetValue` is `NULL`, this method quickly returns true/false depending on
-     * if baggage is set for given key or not. If `targetValue` isn't `NULL` then the
-     * baggage value is assigned to `*targetValue` if it exists, otherwise `*targetValue`
-     * doesn't get modified.
-     *
-     * See the `setBaggageItem()` notes about `restrictedKey`.
-     */
-    virtual bool getBaggageItem(const std::string & restrictedKey, std::string * targetValue = 0) const;
-
-    /**
-     * Provides access to the `Tracer` that created this `Span`. Implementation should
-     * ensure the reference remains valid potentially in a multi-threaded context (might
-     * require some form of reference counting).
-     */
-    virtual const Tracer & getTracer() const = 0;
-
-    /**
-     * Stops this span. `finish()` should be the last call made to any span instance,
-     * and to do otherwise leads to undefined behavior.
-     */
-    virtual void finish();
-
-    /**
-     * Same as `finish()` above but also sets explicitly the finishing time
-     * in microseconds since epoch. If set to the default value (the unix
-	 * epoch), implementations should use the current time implicitly.
-     */
-    virtual void finish(const uint64_t & finishTime);
+    int log(const StringRef& key, const T& val, uint64_t);
+
+    virtual int setBaggage(const StringRef& key, const StringRef& value) = 0;
+
+    virtual int getBaggage(const StringRef& key,
+                           std::string*     baggage) const = 0;
+
+    virtual int getBaggage(const StringRef&          key,
+                           std::vector<std::string>* baggage) const = 0;
+
+    virtual int externalize(std::vector<TextMapPair> *) = 0;
```

Comment:
as discussed here and there, this has intentionally been left out of the OT spec... happy to discuss again if you want :)